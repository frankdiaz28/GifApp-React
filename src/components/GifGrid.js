import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifts } from '../hooks/useFetchGifts';
import GifGridItem from './GifGridItem';

const GifGrid = ({ category }) => {

     const { data:images, loading } = useFetchGifts( category );

    return (
        <>
        <h3 className='animate__fadeIn'>{ category }</h3>

        {loading && <p>Loading</p>}

        <div className='card-grid'>
            {
                images.map( img =>(
                    <GifGridItem 
                      key={img.id}
                      {...img}
                    />
                       
                ))
            }
        </div>
        </>
    )
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid;
