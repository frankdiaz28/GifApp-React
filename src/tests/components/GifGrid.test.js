import React from 'react';
import { shallow } from "enzyme";
import GifGrid from '../../components/GifGrid';
import { useFetchGifts } from '../../hooks/useFetchGifts';
jest.mock('../../hooks/useFetchGifts');

describe('Pruebas en el componente GifGrid', () => {

    const category = 'Mr Robot';
    //const wrapper = shallow( <GifGrid category = { category }/>);

    test('Debe mostar el componente correctamente', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/holamundo/mundo.jpg',
            title: 'HolaMundo'
        }]

        useFetchGifts.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        expect( wrapper ).toMatchSnapshot();
    });
    
});
