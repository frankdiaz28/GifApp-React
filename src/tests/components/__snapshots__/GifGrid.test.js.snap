// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid Debe mostar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__fadeIn"
  >
    Mr Robot
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="HolaMundo"
      url="https://localhost/holamundo/mundo.jpg"
    />
  </div>
</Fragment>
`;
